### ISSUE TITLE
Having non-root privileges on the host and root inside the container

### ISSUE THREAD

**Issue by pwaller** [docker/docker#2918]
This issue is for discussion of one of the two halves of #1034 ("taking advantage of `CONFIG_USER_NS`").

With user namespaces, docker could have ordinary user privileges on the host but _think that they have root privileges inside_ the container.

xref https://github.com/dotcloud/docker/issues/503#issuecomment-26663193

**Comment by jfrazelle** [comment]
duplicate of https://github.com/docker/docker/issues/7906

**Comment by tphyahoo** [comment]
Could this have label project/security added?

**Comment by damienmg** [comment]
I don't see how this is a duplicate of #7906 though there are some connection.

I try to do exactly that, run a docker daemon inside a namespace sandbox and it does not work. With docker 10rc2, I get the following log:

```
INFO[0000] Graph migration to content-addressability took 0.00 seconds 
WARN[0000] Running modprobe bridge br_netfilter failed with message: modprobe: ERROR: could not insert 'bridge': Operation not permitted
modprobe: WARNING: Module br_netfilter not found.
insmod /lib/modules/3.13.0-74-generic/kernel/net/llc/llc.ko 
, error: exit status 1 
WARN[0000] Running modprobe nf_nat failed with message: `modprobe: ERROR: could not insert 'nf_nat': Operation not permitted
insmod /lib/modules/3.13.0-74-generic/kernel/net/netfilter/nf_conntrack.ko`, error: exit status 1 
INFO[0000] Default bridge (docker0) is assigned with an IP address 172.17.0.0/16. Daemon option --bip can be used to set a preferred IP address 
WARN[0000] Your kernel does not support cgroup memory limit: mountpoint for memory not found 
WARN[0000] mountpoint for cpu not found                 
WARN[0000] mountpoint for blkio not found               
WARN[0000] mountpoint for cpuset not found              
FATA[0000] Error starting daemon: Devices cgroup isn't mounted 
```

You can obviously run a docker daemon inside a docker container in priviledged mode but that does not help in running the docker daemon without requiring root privilege.

If that bug is really resolved, is there any description somewhere on how to do it?

**Comment by PAStheLoD** [comment]
@damienmg What happens if you [mount a cgroups hierarchy](https://github.com/tianon/cgroupfs-mount/blob/master/cgroupfs-mount) before starting docker?

**Comment by damienmg** [comment]
Same thing, wether I do the mount inside or outside the namespace sandbox

**Comment by bittner** [comment]
Sorry in advance for a maybe stupid question:

For what I understand reading this discussion and #15187 with the advent of user namespaces in Docker 1.9 the problem of having files owned by `root` on the host file system should be a matter of the past.

## Files still owned by root on a mounted volume

Now, I run docker-engine 1.11.2 and docker-compose 1.7.1. And the files that the Docker container of my (Python) application generates in the mounted volume of the container (e.g. `.pyc` files) are still all owned by `root` on my host system. Which is a bit of a pain when I want to clean up the project and have the intermediate build files, etc. removed from the project. (I have to use `sudo` to first `chown` them or delete them using `sudo` straight away. Feels a bit wacky.)

Is this a different issue, or am I missing something?

**Comment by justincormack** [comment]
@bittner have you enabled user namespaces? They are not enabled by default. However, they do not yet fix all the issues with ownership, there is still work going on upstream in the kernel to make a better solution.
